src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+  health = 100;
 
-class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (this instanceof Carnivore && animal.hidden === false) {
```

Comment:
The check `animal.hidden === false` is not sufficient. You need to ensure that the `animal` is an instance of `Herbivore` before attempting to bite. Use `animal instanceof Herbivore` to check the type of the animal.